gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/AllergyIntoleranceMapper.java

```diff
@@ -124,39 +150,45 @@ private void buildCode(AllergyIntolerance allergyIntolerance, RCMRMT030101UK04Co
                 return;
             }
 
-            DegradedCodeableConcepts.addDegradedEntryIfRequired(
-                    allergyIntolerance.getCode(),
-                    DegradedCodeableConcepts.DEGRADED_NON_DRUG_ALLERGY);
+            DegradedCodeableConcepts.addDegradedEntryIfRequired(allergyIntolerance.getCode(),
+                DegradedCodeableConcepts.DEGRADED_NON_DRUG_ALLERGY);
         }
 
         if (DRUG_ALLERGY_CODE.equals(compoundCode)) {
-            if (observationStatement.hasValue() && observationStatement.getValue() instanceof CD value) {
-                if (!CODE_SYSTEM.equals(value.getCodeSystem())) {
-                    var codeableConceptFromValue = codeableConceptMapper.mapToCodeableConcept(value);
-                    allergyIntolerance.setCode(codeableConceptFromValue);
-
-                    var codeDisplayName = codeableConceptFromCode.getCodingFirstRep().getDisplay();
-                    if (codeDisplayName != null && !ALLERGY_TERM_TEXT.equals(codeDisplayName)
-                        && !codeDisplayName.equals(codeableConceptFromValue.getCodingFirstRep().getDisplay())) {
-                        allergyIntolerance.getNote().add(new Annotation().setText(ALLERGY_NOTE.formatted(codeDisplayName)));
-                    }
-                } else {
-                    allergyIntolerance.setCode(codeableConceptFromCode);
-                }
-            } else {
-                allergyIntolerance.setCode(codeableConceptFromCode);
-            }
-
-            if (allergyIntolerance.getCode() == null) {
+            var codeableConceptFromValue = getCodeableConceptFromNonEgtonCodeValue(observationStatement);
+            if (codeableConceptFromValue == null && codeableConceptFromCode == null) {
                 return;
             }
+            if (codeableConceptFromValue == null) {
+                allergyIntolerance.setCode(codeableConceptFromCode);
 
-            DegradedCodeableConcepts.addDegradedEntryIfRequired(
+                buildAllergyIntoleranceText(observationStatement, allergyIntolerance);
+
+                DegradedCodeableConcepts.addDegradedEntryIfRequired(
                     allergyIntolerance.getCode(),
                     DegradedCodeableConcepts.DEGRADED_DRUG_ALLERGY);
+                return;
+            }
+
+            allergyIntolerance.setCode(codeableConceptFromValue);
+
+            var allergyNote = getAllergyNote(codeableConceptFromCode, codeableConceptFromValue);
+            if (allergyNote != null) {
+                allergyIntolerance.getNote().add(allergyNote);
+            }
+        } else {
+            allergyIntolerance.setCode(codeableConceptFromCode);
```

Comment:
From my understanding `mapToAllergyIntollerence` wouldn't be called if the compound code was not  `SN53.00` or `14L..00`  so we could throw an exception here (or do nothing).

Reply:
This seems be correct, I don't believe we need this final else